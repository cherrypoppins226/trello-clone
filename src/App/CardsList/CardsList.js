import React from "react";
import { withStyles } from "@material-ui/core/styles";
import Button from "@material-ui/core/Button";
import Paper from "@material-ui/core/Paper";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { fileAbsolute } from "paths.macro";

import Header from "./Header";
import Cards from "./Cards";
import { button } from "../styles";
import { moduleName } from "../utils";
import { mapDispatchToProps } from "../redux";

const styles = {
  root: {
    display: "flex",
    flexDirection: "column",
    background: "rgb(226, 228, 230)",
    "& > button": {
      ...button,
      justifyContent: "left",
      fontWeight: 400
    }
  },
  header: {
    margin: 5
  }
};

const View = props => {
  return (
    <Paper
      data-listid={props.list.id}
      component="section"
      elevation={1}
      className={props.classes.root}
    >
      <Header
        className={props.classes.header}
        listId={props.list.id}
        listTitle={props.list.title}
      />
      <div style={{ overflowY: "scroll" }}>
        <Cards cards={props.list.cards} />
      </div>
      <Button onClick={() => props.actions.addCard(props.list.id)}>
        Add a card...
      </Button>
    </Paper>
  );
};

const mapStateToProps = (state, ownProps) => ({
  list: state.lists.find(list => list.id === ownProps.list.id)
});

const Container = connect(mapStateToProps, mapDispatchToProps)(
  withStyles(styles)(View)
);

Container.displayName = moduleName(fileAbsolute);

export const listType = PropTypes.shape({
  id: PropTypes.number.isRequired,
  title: PropTypes.string.isRequired,
  cards: PropTypes.array.isRequired
});

Container.propTypes = {
  list: listType
};

export default Container;
