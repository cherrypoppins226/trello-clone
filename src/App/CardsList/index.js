export { default, listType } from "./CardsList";
