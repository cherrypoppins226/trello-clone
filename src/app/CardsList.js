import React from "react";
import { withStyles } from "@material-ui/core/styles";
import Button from "@material-ui/core/Button";
import Paper from "@material-ui/core/Paper";
import PropTypes from "prop-types";
import { fileAbsolute } from "paths.macro";
import { graphql } from "react-apollo";
import { compose, setDisplayName, setPropTypes } from "recompose";

import AppState from "../App.state";
import Header from "./cardsList/Header";
import Cards from "./cardsList/Cards";
import { queries } from "../cosmos/apollo/schema";
import { button } from "./styles";
import { makeFixtures, handleGraphQLResponse, moduleName } from "../utils";

const styles = {
  root: {
    display: "flex",
    flexDirection: "column",
    background: "rgb(226, 228, 230)",
    "& > button": {
      ...button,
      justifyContent: "left",
      fontWeight: 400
    }
  },
  header: {
    margin: 5
  }
};

const CardsList = ({ classes, id, addCard, data: { list, variables } }) => {
  return (
    <Paper
      data-listid={id}
      component="section"
      elevation={1}
      className={classes.root}
    >
      <Header
        className={classes.header}
        listId={list.id}
        listTitle={list.title}
      />
      <div style={{ overflowY: "scroll" }}>
        <Cards cards={list.cards} />
      </div>
      <Button
        onClick={() => {
          const listQuery = { query: queries.list, variables };
          addCard({
            update: (proxy, { data: { addCard } }) => {
              const data = proxy.readQuery(listQuery);
              data.list.cards.push(addCard);
              proxy.writeQuery({ ...listQuery, data });
            }
          });
        }}
      >
        Add a card...
      </Button>
    </Paper>
  );
};

const Component = compose(
  setDisplayName(moduleName(fileAbsolute)),
  setPropTypes({
    id: PropTypes.number.isRequired
  }),
  graphql(queries.list, {
    options: props => ({ variables: { id: props.id } })
  }),
  handleGraphQLResponse(),
  graphql(queries.addCard, {
    name: "addCard",
    options: props => ({
      variables: { listId: props.id, title: "Title..." },
      optimisticResponse: {
        __typename: "Mutation",
        addCard: {
          __typename: "Card",
          id: -1,
          title: "Title..."
        }
      }
    })
  }),
  withStyles(styles)
)(CardsList);

export const fixtures = makeFixtures(Component, {
  default: {
    props: {
      id: 1
    },
    stores: {
      appState: new AppState()
    }
  }
});

export default Component;
