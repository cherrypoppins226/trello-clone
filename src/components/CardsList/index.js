import React from "react";
import { withStyles } from "material-ui/styles";
import Button from "material-ui/Button";
import Paper from "material-ui/Paper";
import grey from "material-ui/colors/grey";
import PropTypes from "prop-types";
import MoreHoriz from "@material-ui/icons/MoreHoriz";
import Title from "./Title";
import Card from "./Card";
import * as Labels from "../labels";

const styles = {
  root: {
    display: "flex",
    flexDirection: "column",
    maxHeight: "100%",
    background: "rgb(226, 228, 230)",
    "& > button": {
      justifyContent: "left",
      textTransform: "none"
    },
    "& ul": {
      padding: 0,
      margin: 0,
      height: "100%",
      "& li": {
        margin: 8,
        marginTop: 2,
        cursor: "pointer"
      }
    }
  },
  listHeader: {
    display: "flex",
    flexShrink: 0,
    justifyContent: "space-between",
    margin: 5,
    "& [role='heading']": {
      flexBasis: "100%",
      fontWeight: 700,
      paddingBottom: 0,
      paddingTop: 3,
      paddingLeft: 5,
      marginRight: 8
    },
    "& p[role='heading']": {
      cursor: "pointer"
    },
    "& button": {
      alignSelf: "flex-start",
      // Reusable styles
      borderWidth: 0,
      background: "none",
      outline: "none",
      cursor: "pointer",
      borderRadius: 4,
      padding: 3,
      "&:hover": {
        background: "#D6D6D6"
      }
    },
    "& svg": {
      // Reusable styles
      padding: 3,
      width: "0.8em",
      height: "0.8em",
      color: grey[700]
    }
  }
};

class CardsList extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      counter: props.cards.length,
      cards: props.cards.map((card, idx) => ({
        id: idx,
        description: card
      }))
    };
  }

  addCard(e) {
    e.preventDefault();
    this.setState((prevState, props) => ({
      counter: prevState.counter + 1,
      cards: [
        ...prevState.cards,
        { id: prevState.counter + 1, description: undefined }
      ]
    }));
  }

  componentDidUpdate(prevProps, prevState) {
    if (prevState.cards.length < this.state.cards.length)
      this.cardsListEnd.scrollIntoView();
  }

  render() {
    const { classes } = this.props;
    return (
      <Paper
        component="section"
        elevation={1}
        className={classes.root}
        data-testid="CardsList"
      >
        <div className={classes.listHeader}>
          <Title role="heading" text={this.props.title} />
          <button
            onClick={e => this.props.onEditList(e.currentTarget)}
            aria-labelledby={Labels.cardsListActionsMenu.id}
            aria-haspopup={true}
          >
            <MoreHoriz />
          </button>
        </div>
        <div style={{ overflowY: "scroll" }}>
          <ul>
            {this.state.cards.map(({ id, description }) => (
              <li key={id}>
                <Card
                  description={description}
                  onQuickEditCard={this.props.onQuickEditCard}
                  onEditCard={this.props.onEditCard}
                />
              </li>
            ))}
          </ul>
          {/* The sole purpose of this div is to allow scrolling to bottom */}
          <div
            style={{ float: "left", clear: "both" }}
            ref={node => (this.cardsListEnd = node)}
          />
        </div>
        <Button onClick={this.addCard.bind(this)}>Add a card...</Button>
      </Paper>
    );
  }
}

const View = withStyles(styles)(CardsList);

View.propTypes = {
  title: PropTypes.string.isRequired,
  cards: PropTypes.array.isRequired,
  onEditList: PropTypes.func.isRequired,
  onQuickEditCard: PropTypes.func.isRequired,
  onEditCard: PropTypes.func.isRequired
};

export default View;
