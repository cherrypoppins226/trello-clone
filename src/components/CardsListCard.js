import React from "react";
import path from "path";
import { withStyles } from "material-ui/styles";
import Paper from "material-ui/Paper";
import Typography from "material-ui/Typography";
import PropTypes from "prop-types";
import grey from "material-ui/colors/grey";
import Create from "@material-ui/icons/Create";
import * as Labels from "./labels";

const styles = {
  container: {
    margin: 8,
    marginTop: 2,
    cursor: "pointer"
  },
  card: {
    padding: 4,
    paddingLeft: 8,
    display: "flex",
    flexDirection: "row",
    justifyContent: "space-between",
    "&:hover": {
      background: grey[200]
    },
    "&:hover button": {
      visibility: "visible"
    },
    "& button": {
      visibility: "hidden",
      // Reusable styles
      borderWidth: 0,
      background: "none",
      outline: "none",
      cursor: "pointer",
      alignSelf: "flex-start",
      borderRadius: 4,
      padding: 3,
      paddingBottom: 0,
      "&:hover": {
        background: "#D6DADC"
      }
    },
    "& svg": {
      // Reusable styles
      padding: 3,
      width: "0.8em",
      height: "0.8em",
      color: grey[700]
    }
  }
};

const onClick = (event, onEditCard, onEditFullCard) => {
  const button = event.currentTarget.querySelector(
    `[aria-labelledby="${Labels.editCard.id}"]`
  );
  if (button.contains(event.target)) {
    onEditCard(event.currentTarget);
  } else {
    onEditFullCard(event.currentTarget);
  }
};

export const cardDescription = cardNode => {
  return cardNode.querySelector(`[aria-labelledby="${Labels.card.id}"]`)
    .textContent;
};

const View = ({
  classes,
  description = "Title...",
  onEditCard,
  onEditFullCard
}) => {
  return (
    <li data-testid="CardsListCard" className={classes.container}>
      <Paper
        elevation={1}
        onClick={e => onClick(e, onEditCard, onEditFullCard)}
        className={classes.card}
        aria-labelledby={Labels.fullyEditCard.id}
      >
        <Typography aria-labelledby={Labels.card.id}>{description}</Typography>
        <button aria-labelledby={Labels.editCard.id}>
          <Create />
        </button>
      </Paper>
    </li>
  );
};

View.propTypes = {
  children: PropTypes.node,
  classes: PropTypes.object.isRequired,
  description: PropTypes.string,
  onEditCard: PropTypes.func.isRequired,
  onEditFullCard: PropTypes.func.isRequired
};

View.displayName = path.basename(__filename, path.extname(__filename));

export default withStyles(styles)(View);
