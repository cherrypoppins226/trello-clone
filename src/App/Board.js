import React from "react";
import { withStyles } from "@material-ui/core/styles";
import Grid from "@material-ui/core/Grid";
import Modal from "@material-ui/core/Modal";
import PropTypes from "prop-types";
import Popover from "@material-ui/core/Popover";
import CardsList from "./CardsList";
import QuickEditCard from "./QuickEditCard";
import ActionsMenu from "./CardsList/ActionsMenu";
import EditCard from "./EditCard";
import { cardDescription } from "./CardsList/Card";

const styles = {
  root: {
    padding: 8,
    margin: 0,
    height: "100%",
    width: "100%"
  },
  cardListContainer: {
    width: 280
  },
  cardsListActionsPopover: {
    backgroundColor: "transparent"
  }
};

class Board extends React.Component {
  state = {
    listBeingEdited: null,
    cardBeingQuickEdited: null,
    cardBeingEdited: null
  };

  render() {
    const { classes, lists } = this.props;
    const cardTitle = (v => (v ? cardDescription(v) : ""))(
      this.state.cardBeingEdited || this.state.cardBeingQuickEdited
    );
    return (
      <>
        <Grid container spacing={16} wrap="nowrap" className={classes.root}>
          {Object.entries(lists).map(([k, v]) => {
            return (
              <Grid item key={k} className={classes.cardListContainer}>
                <CardsList
                  title={k}
                  cards={v}
                  onQuickEditCard={card =>
                    this.setState({ cardBeingQuickEdited: card })
                  }
                  onEditCard={card => this.setState({ cardBeingEdited: card })}
                  onEditList={list => this.setState({ listBeingEdited: list })}
                />
              </Grid>
            );
          })}
        </Grid>
        <Popover
          anchorEl={this.state.listBeingEdited}
          container={this}
          open={Boolean(this.state.listBeingEdited)}
          onClose={_ => this.setState({ listBeingEdited: null })}
          TransitionProps={{ timeout: 0 }}
          elevation={1}
          anchorOrigin={{ vertical: "bottom", horizontal: "left" }}
          transformOrigin={{ vertical: "top", horizontal: "left" }}
          PaperProps={{ classes: { root: classes.cardsListActionsPopover } }}
        >
          <ActionsMenu
            onMenuItemClick={_ => this.setState({ listBeingEdited: null })}
          />
        </Popover>
        <Modal
          container={this}
          open={Boolean(this.state.cardBeingQuickEdited)}
          onClose={_ => this.setState({ cardBeingQuickEdited: null })}
          BackdropProps={{ id: "quickEditCardBackdrop" }}
        >
          <QuickEditCard
            title={cardTitle}
            anchorEl={this.state.cardBeingQuickEdited}
          />
        </Modal>
        <Modal
          container={this}
          open={Boolean(this.state.cardBeingEdited)}
          onClose={_ => this.setState({ cardBeingEdited: null })}
          BackdropProps={{ id: "editCardBackdrop" }}
        >
          <EditCard title={cardTitle} />
        </Modal>
      </>
    );
  }
}

const View = withStyles(styles)(Board);

View.propTypes = {
  lists: PropTypes.object.isRequired
};

export default View;
