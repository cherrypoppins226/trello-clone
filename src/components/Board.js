import React from "react";
import { withStyles } from "material-ui/styles";
import Grid from "material-ui/Grid";
import PropTypes from "prop-types";
import Popover from "material-ui/Popover";
import CardsList from "./CardsList";
import QuickEditCard from "./QuickEditCard";
import ActionsMenu from "./CardsList/ActionsMenu";
import EditCard from "./EditCard";
import * as Labels from "./labels";

const styles = {
  root: {
    padding: 8,
    margin: 0,
    height: "100%",
    width: "100%"
  },
  cardListContainer: {
    width: 280
  }
};

class Board extends React.Component {
  state = {
    listBeingEdited: null,
    cardBeingQuickEdited: null,
    cardBeingEdited: null
  };

  render() {
    const { classes, lists } = this.props;
    return (
      <>
        <Grid container spacing={16} wrap="nowrap" className={classes.root}>
          {Object.entries(lists).map(([k, v]) => {
            return (
              <Grid item key={k} className={classes.cardListContainer}>
                <CardsList
                  title={k}
                  cards={v}
                  onQuickEditCard={card =>
                    this.setState({ cardBeingQuickEdited: card })
                  }
                  onEditCard={card => this.setState({ cardBeingEdited: card })}
                  onEditList={list => this.setState({ listBeingEdited: list })}
                />
              </Grid>
            );
          })}
        </Grid>
        <Popover
          aria-describedby={Labels.cardsListActionsMenuDescription.id}
          anchorEl={this.state.listBeingEdited}
          container={this}
          open={Boolean(this.state.listBeingEdited)}
          onClose={_ => this.setState({ listBeingEdited: null })}
          transitionDuration={0}
          elevation={1}
          anchorOrigin={{ vertical: "bottom", horizontal: "left" }}
          transformOrigin={{ vertical: "top", horizontal: "left" }}
        >
          <ActionsMenu
            onMenuItemClick={_ => this.setState({ listBeingEdited: null })}
          />
        </Popover>
        <QuickEditCard
          container={this}
          card={this.state.cardBeingQuickEdited}
          onClose={_ => this.setState({ cardBeingQuickEdited: null })}
        />
        <EditCard
          container={this}
          card={this.state.cardBeingEdited}
          onClose={_ => this.setState({ cardBeingEdited: null })}
        />
      </>
    );
  }
}

const View = withStyles(styles)(Board);

View.propTypes = {
  lists: PropTypes.object.isRequired
};

export default View;
