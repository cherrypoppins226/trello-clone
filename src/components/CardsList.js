import React from "react";
import { findDOMNode } from "react-dom";
import path from "path";
import { withStyles } from "material-ui/styles";
import Button from "material-ui/Button";
import Typography from "material-ui/Typography";
import Paper from "material-ui/Paper";
import grey from "material-ui/colors/grey";
import PropTypes from "prop-types";
import MoreHoriz from "@material-ui/icons/MoreHoriz";
import TextArea from "./TextArea";
import CardsListCard from "./CardsListCard";
import * as Labels from "./labels";

const styles = {
  container: {
    display: "flex",
    flexDirection: "column",
    maxHeight: "100%",
    background: "rgb(226, 228, 230)",
    "& > button": {
      justifyContent: "left",
      textTransform: "none"
    },
    "& ul": {
      padding: 0,
      margin: 0,
      height: "100%"
    }
  },
  listHeader: {
    display: "flex",
    flexShrink: 0,
    justifyContent: "space-between",
    margin: 5,
    "& [role='heading']": {
      flexBasis: "100%",
      fontWeight: 700,
      paddingBottom: 0,
      paddingTop: 3,
      paddingLeft: 5,
      marginRight: 8
    },
    "& p[role='heading']": {
      cursor: "pointer"
    },
    "& button": {
      alignSelf: "flex-start",
      // Reusable styles
      borderWidth: 0,
      background: "none",
      outline: "none",
      cursor: "pointer",
      borderRadius: 4,
      padding: 3,
      "&:hover": {
        background: "#D6D6D6"
      }
    },
    "& svg": {
      // Reusable styles
      padding: 3,
      width: "0.8em",
      height: "0.8em",
      color: grey[700]
    }
  }
};

const onOutsideClick = (node, handler) => {
  const onClick = event => {
    if (!node.contains(event.target)) {
      document.body.removeEventListener("click", onClick);
      handler(event);
    }
  };
  document.body.addEventListener("click", onClick);
};

const Title = class extends React.Component {
  constructor(props) {
    super(props);
    this.state = { editing: false };
  }

  componentDidUpdate(prevProps, prevState) {
    if (this.state.editing)
      onOutsideClick(findDOMNode(this), _ => this.setState({ editing: false }));
  }

  render() {
    const props = this.state.editing
      ? {
          component: TextArea,
          value: this.props.text
        }
      : {
          onClick: _ => this.setState({ editing: true }),
          children: this.props.text
        };
    return <Typography role="heading" {...props} />;
  }
};

const View = class extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      counter: props.cards.length,
      cards: props.cards.map((description, idx) =>
        this.newCard(idx, description)
      )
    };
  }

  newCard(id, description = undefined) {
    return (
      <CardsListCard
        key={id}
        description={description}
        onEditCard={this.props.onEditCard}
      />
    );
  }

  addCard(e) {
    e.preventDefault();
    this.setState((prevState, props) => ({
      counter: prevState.counter + 1,
      cards: [...prevState.cards, this.newCard(prevState.counter + 1)]
    }));
  }

  componentDidUpdate(prevProps, prevState) {
    if (prevState.cards.length < this.state.cards.length)
      this.cardsListEnd.scrollIntoView();
  }

  render() {
    const { classes } = this.props;
    return (
      <Paper
        component="section"
        elevation={1}
        className={classes.container}
        data-testid="CardsList"
      >
        <div className={classes.listHeader}>
          <Title text={this.props.title} />
          <button
            onClick={e => this.props.onEditList(e.currentTarget)}
            aria-describedby={Labels.cardsListActionsMenu.id}
            aria-haspopup={true}
          >
            <MoreHoriz />
          </button>
        </div>
        <div style={{ overflowY: "scroll" }}>
          <ul>{this.state.cards}</ul>
          {/* The sole purpose of this div is to allow scrolling to bottom */}
          <div
            style={{ float: "left", clear: "both" }}
            ref={node => (this.cardsListEnd = node)}
          />
        </div>
        <Button onClick={this.addCard.bind(this)}>Add a card...</Button>
      </Paper>
    );
  }
};

View.propTypes = {
  children: PropTypes.node,
  classes: PropTypes.object.isRequired,
  title: PropTypes.string.isRequired,
  cards: PropTypes.array.isRequired,
  onEditList: PropTypes.func.isRequired,
  onEditCard: PropTypes.func.isRequired
};

View.displayName = path.basename(__filename, path.extname(__filename));

export default withStyles(styles)(View);
