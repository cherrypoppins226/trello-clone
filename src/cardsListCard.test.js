import React from "react";
import faker from "faker";
import { Simulate } from "react-testing-library";
import { render } from "./testHelpers.js";
import CardsList from "./cardsList";
import CardsListCard, {
  EDIT_CARD_LABEL,
  cardDescription
} from "./cardsListCard";
import Board, { EDIT_CARD_DESCRIPTION } from "./board";

it("initial render", () => {
  const { container } = render(
    <CardsListCard description="Card list card text" onEditCard={jest.fn()} />
  );
  expect(container).toMatchSnapshot();
});

it("adds a card", () => {
  const { getByText, getByTestId } = render(
    <CardsList
      title="Title"
      cards={["card1", "card2"]}
      onEditCard={jest.fn()}
    />
  );
  const liveList = getByTestId("cards-list");
  const countBefore = liveList.childElementCount;
  const lastBefore = liveList.lastElementChild;
  Simulate.click(getByText("Add a card..."));
  expect(liveList.childElementCount).toBe(countBefore + 1);
  expect(liveList.lastElementChild).not.toBe(lastBefore);
  expect(liveList.lastElementChild.tagName).toBe(lastBefore.tagName);
});

faker.seed(1);

const lists = {
  [faker.lorem.sentence()]: [faker.lorem.sentence(), faker.lorem.sentence()],
  [faker.lorem.sentence()]: [faker.lorem.sentence(), faker.lorem.sentence()]
};

it("opens edit card modal with correct text", () => {
  const { container } = render(<Board lists={lists} />);
  const editCard = container.querySelector(
    `[aria-labelledby="${EDIT_CARD_LABEL}"]`
  );
  Simulate.click(editCard);
  const modal = container.querySelector(
    `[aria-describedby="${EDIT_CARD_DESCRIPTION}"]`
  );
  expect(modal).not.toBeNull();
  const modalContainer = editCard.parentElement;
  expect(modal.parentElement).toBe(modalContainer);
  expect(modal.querySelector("textarea").textContent).toBe(
    cardDescription(modalContainer)
  );
});
