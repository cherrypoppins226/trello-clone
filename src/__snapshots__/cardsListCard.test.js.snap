// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`initial render 1`] = `
<div>
  <li
    class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded View-container-1"
  >
    <div
      class="MuiButtonBase-root MuiListItem-root MuiListItem-default MuiListItem-gutters MuiListItem-button View-card-2"
      role="button"
      tabindex="0"
    >
      <p
        aria-label="card title"
        class="MuiTypography-root MuiTypography-body1"
      >
        Card list card text
      </p>
      <div
        id="edit-card-label"
        style="display: none;"
      >
        Edit card
      </div>
      <button
        aria-labelledby="edit-card-label"
        class="View-edit-3"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root View-editIcon-4"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <g>
            <path
              d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34c-.39-.39-1.02-.39-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
            />
          </g>
        </svg>
      </button>
      <span
        class="MuiTouchRipple-root"
      />
    </div>
  </li>
</div>
`;
